File: ontrack-extension-jenkins/src/main/java/net/nemerosa/ontrack/extension/jenkins/AbstractJenkinsProperty.java
package net.nemerosa.ontrack.extension.jenkins;

import lombok.Data;
import net.nemerosa.ontrack.model.support.ConfigurationProperty;

@Data
public abstract class AbstractJenkinsProperty implements ConfigurationProperty<JenkinsConfiguration> {

    /**
     * Reference to the Jenkins configuration.
     */
    private final JenkinsConfiguration configuration;

}

File: ontrack-extension-jenkins/src/main/java/net/nemerosa/ontrack/extension/jenkins/AbstractJenkinsProperty.kt
package net.nemerosa.ontrack.extension.jenkins

import net.nemerosa.ontrack.model.docs.DocumentationType
import net.nemerosa.ontrack.model.support.ConfigurationProperty

/**
 * Property linked to a Jenkins configuration.
 *
 * @property configuration Reference to the Jenkins configuration.
 */
abstract class AbstractJenkinsProperty(
    @DocumentationType("String", "Name of the Jenkins configuration")
    override val configuration: JenkinsConfiguration
) : ConfigurationProperty<JenkinsConfiguration>
